hazelcast-client/src/main/java/com/hazelcast/client/impl/clientside/FailoverClientConfigSupport.java

```diff
@@ -38,41 +43,85 @@ private FailoverClientConfigSupport() {
+     * failover configuration using the search logic described in
+     * {@link HazelcastClient#newHazelcastFailoverClient()} and builds
+     * {@link ClientFailoverConfig}.
+     *
+     * @param clientFailoverConfig provided via {@link HazelcastClient#newHazelcastFailoverClient(ClientFailoverConfig)}
+     * @return resolvedConfigs
+     * @throws HazelcastException  if {@code clientFailoverConfig} is {@code null} and no failover configuration is found
+     * @throws InvalidConfigurationException when given config is not valid
+     */
+    public static ClientFailoverConfig resolveClientFailoverConfig(ClientFailoverConfig clientFailoverConfig) {
         if (clientFailoverConfig == null) {
-            XmlClientFailoverConfigBuilder configBuilder = new XmlClientFailoverConfigBuilder();
-            clientFailoverConfig = configBuilder.build();
+            clientFailoverConfig = locateAndCreateClientFailoverConfig();
         }
         checkValidAlternative(clientFailoverConfig.getClientConfigs());
         return clientFailoverConfig;
     }
 
     /**
-     * Returns a ClientFailoverConfig with single client config. If clientConfig is null,
-     * We create it via XmlClientConfigBuilder().build()
+     * Returns a {@link ClientFailoverConfig} with single client config.
+     * If the provided {@code clientConfig} is {@code null}, client
+     * configuration is loaded using the resolution logic as described in
+     * {@link HazelcastClient#newHazelcastClient()}
      *
-     * @param config provided via HazelcastClient.newHazelcastClient(ClientConfig config)
+     * @param config provided via {@link HazelcastClient#newHazelcastClient(ClientConfig)}
      * @return resolvedConfigs
      * @throws InvalidConfigurationException when given config is not valid
      */
     public static ClientFailoverConfig resolveClientConfig(ClientConfig config) {
         if (config == null) {
-            config = createDefaultClientConfig();
+            config = locateAndCreateClientConfig();
         }
         ClientFailoverConfig resolvedConfig = new ClientFailoverConfig();
         resolvedConfig.addClientConfig(config);
         resolvedConfig.setTryCount(1);
         return resolvedConfig;
     }
 
-    private static ClientConfig createDefaultClientConfig() {
+    private static ClientFailoverConfig locateAndCreateClientFailoverConfig() {
+        ClientFailoverConfig config;
+        XmlClientFailoverConfigLocator xmlConfigLocator = new XmlClientFailoverConfigLocator();
+        YamlClientFailoverConfigLocator yamlConfigLocator = new YamlClientFailoverConfigLocator();
+
+        if (xmlConfigLocator.locateFromSystemProperty()) {
+            // 1. Try loading XML config if provided in system property
+            config = new XmlClientFailoverConfigBuilder(xmlConfigLocator).build();
+
+        } else if (yamlConfigLocator.locateFromSystemProperty()) {
+            // 2. Try loading YAML config if provided in system property
```

Comment:
The order here is different from what was documented above in javadoc at class `HazelcastClient.java`.

Reply:
It tries loading the file referenced in the system property. If it is an xml file, the xml locator will find it, if it is a yaml file, the yaml locator will find it. But since the system property references a single file, there is no particular ordering here. The two conditions/locator calls can be written in the opposite order without behavior change. The locators check the extension of the file to determine whether it is an xml or a yaml file.

Is it what you meant?

Reply:
See `HazelcastClient.newHazelcastFailoverClient()` javadoc.

Reply:
Extended the javadoc of the `HazelcastClient` methods to clarify system property can be an XML or a YAML file that is distinguished by the suffix of the file name.

Reply:
Changed the inline comments slightly. I wouldn't change the referenced javadoc, since it is an implementation detail that we try loading XML and YAML in two steps and it does not affect the search order. If an XML or a YAML file is configured in the system property, we load from the system property and not from lower priority locations (classpath or workdir).